File: JS/producer_v2.js
function* producer() {
    yield "data1";
    yield "data3";
}

function* producer_2() {
    yield "data2";
}

const prod = producer();
const prod2 = producer_2();

console.log(prod2.next().value);


for (const value of prod) console.log(value);

File: Lua/producer_v2.lua
function producer()
    coroutine.yield("data1")
    coroutine.yield("data3")
    end

function producer_2()
    coroutine.yield("data2")
end

co1 = coroutine.create(producer)
co2 = coroutine.create(producer_2)

print(coroutine.resume(co1))

-- status, value = coroutine.resume(co1)  -- "data1"
-- print(status, value)  -- true, data1

-- status, value = coroutine.resume(co2)  -- "data2"
-- print(status, value)  -- true, data2

-- status, value = coroutine.resume(co1)  -- "data3"
-- print(status, value)  -- true, data3

